rust/src/encodings/plain.rs

```diff
@@ -0,0 +1,91 @@
+//  Copyright 2022 Lance Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+//! Plain encoding
+
+use std::io::{Error, ErrorKind, Read, Result, Seek, SeekFrom};
+use std::sync::Arc;
+
+use arrow2::array::{Array, MutablePrimitiveArray};
+use arrow2::array::new_empty_array as new_empty_array2;
+use arrow2::array::Int32Array as Int32Array2;
+use arrow2::compute::arithmetics::basic::sub_scalar;
+use arrow2::compute::take::take;
+use arrow2::types::NativeType;
+
+use crate::encodings::Decoder;
+
+/// Plain Decoder
+pub struct PlainDecoder<'a, R: Read + Seek> {
+    file: &'a mut R,
+    position: u64,
+    page_length: i64,
+}
+
+impl<'a, R: Read + Seek> PlainDecoder<'a, R> {
+    pub fn new(file: &'a mut R, position: u64, page_length: i64) -> Self {
+        PlainDecoder {
+            file,
+            position,
+            page_length,
+        }
+    }
+}
+
+impl<'a, R: Read + Seek, T: NativeType> Decoder<T> for PlainDecoder<'a, R> {
+    type ArrowType = T;
+
+    fn decode(&mut self, offset: i32, length: &Option<i32>) -> Result<Arc<dyn Array>> {
+        let read_len = length.unwrap_or((self.page_length - (offset as i64)) as i32) as usize;
+        (*self.file).seek(SeekFrom::Start(self.position + offset as u64))?;
+        // let mut mutable_buf = Buffer::new(read_len * T::get_byte_width());
+        let byte_size = std::mem::size_of::<T>();
+        let mut buf = vec![0u8; read_len * byte_size];//TODO is it right?
```

Comment:
These lines are obviously different from @eddyxu 's `arrow-rs` version because there are missing equivalent things in `arrow2`, need deep review.

Reply:
T is native rust type? Does it support bool?

Reply:
T is an [Arrow2 trait](https://github.com/jorgecarleitao/arrow2/blob/a1ab5a7ea53517717415f3780447ab11e30c5c49/src/types/native.rs#L11) that extends native rust Type. From [this list](https://github.com/jorgecarleitao/arrow2/blob/a1ab5a7ea53517717415f3780447ab11e30c5c49/src/types/native.rs#L76-L86) I think bool is not included, but maybe bool is just UInt8?